### PR TITLE
Update ImageGallery toolbar: changing icon to text and adding ellipsis menu

### LINKED ISSUE #38954: [Fix] Replace the 'Set as cover' icon button with a text button
Usability testing has shown that people struggle to find the 'Set as cover' button which we currently show as an icon in the image toolbar. Let's change it to a regular text button to increase its visibility.

**Current state**

image

**Desired state**

image

**Acceptance criteria**

- [ ] 'Set as cover' is displayed as a text button, not an icon.
- [ ] To clean up the toolbar, let's move `Replace` to the `...` dropdown.

### PR DESCRIPTION
**Pull request by louwie17**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Update ImageGallery toolbar: changing icon to text and adding ellipsis menu

Closes #38954 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Load this branch and go to **Settings > Advanced > Features** and enable the **New Product Editor**
2. Click **Products > Add New** and scroll down to the **Images** field
3. Click **Choose images** and select several images ( you can hold shift to select several )
4. Once added, click on a none-cover image. Notice how the toolbar has a `Set as cover` button
5. Click the `Set as cover` button, and it should replace the first (cover ) image with this one.
6. Now click on another none-cover image and click the 3-dot menu
7. You should see the Replace and Remove options
8. Click **Replace** and select an image to see if it still correctly replaces the image
9. Click **Remove** to make sure it still correctly removes the image.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/components/src/image-gallery/image-gallery-toolbar-dropdown.scss [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// Hack to hide when the media modal is open
+// Otherwise in Firefox the popover remains visible on top of the modal
+// Changing the z-index of Popovers have wider implications.
+.modal-open .woocommerce-image-gallery__toolbar-dropdown-popover {
+	display: none;
+}
```
Comment (louwie17) [inline packages/js/components/src/image-gallery/image-gallery-toolbar-dropdown.scss:6]: See comment above, this follows inline with how Gutenberg does it as well.

File: packages/js/components/src/image-gallery/image-gallery-toolbar.scss [hunk 2]
```diff
@@ -1,4 +1,5 @@
 .woocommerce-image-gallery__toolbar {
+	width: max-content;
```
Comment (louwie17) [inline packages/js/components/src/image-gallery/image-gallery-toolbar.scss:2]: Fixes an issue with the toolbar width not adjusting correctly previously.

File: packages/js/components/src/image-gallery/image-gallery.tsx [hunk 3]
```diff
@@ -129,6 +129,18 @@ export const ImageGallery: React.FC< ImageGalleryProps > = ( {
...
+										(
+											event.relatedTarget as Element
+										 ).closest(
+											'.woocommerce-image-gallery__toolbar'
+										) ) ||
+									( event.relatedTarget &&
+										(
+											event.relatedTarget as Element
+										 ).closest(
+											'.woocommerce-image-gallery__toolbar-dropdown-popover'
```
Comment (louwie17) [inline packages/js/components/src/image-gallery/image-gallery.tsx:145]: A bit dirty, but this was needed to allow us to open the additional dropdown without it auto closing.

File: packages/js/components/src/image-gallery/image-gallery-toolbar-dropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+						<MediaUploadComponent
+							onSelect={ ( media ) => {
+								onReplace( media as MediaItem );
+								onClose();
+							} }
+							allowedTypes={ [ 'image' ] }
+							render={ ( { open } ) => (
+								<MenuItem
+									onClick={ () => {
+										// onClose();
```
Comment (nathanss) [inline packages/js/components/src/image-gallery/image-gallery-toolbar-dropdown.tsx]: Oops!

File: packages/js/components/src/image-gallery/image-gallery.tsx (hunk 3 above)
Comment (nathanss) [reply packages/js/components/src/image-gallery/image-gallery.tsx:145]: I think it would be valuable to leave a comment here explaining the reasoning

**Review by nathanss — CHANGES_REQUESTED** [review]

File: packages/js/components/src/image-gallery/image-gallery-toolbar-dropdown.tsx (hunk 4 above)
Comment (louwie17) [reply packages/js/components/src/image-gallery/image-gallery-toolbar-dropdown.tsx]: Removed in: https://github.com/woocommerce/woocommerce/pull/39753/commits/d682422d93e8577608365354f17d82306412c774

**Comment by louwie17** [comment]
Thanks for the review @nathanss, this should be good for a re-review, I addressed your comments.

**Review by nathanss — APPROVED** [review]

**Pull request merged**